hedera-node/src/main/java/com/hedera/services/txns/crypto/TransferLogic.java

```diff
@@ -0,0 +1,88 @@
+package com.hedera.services.txns.crypto;
+
+import com.hedera.services.context.properties.GlobalDynamicProperties;
+import com.hedera.services.grpc.marshalling.ImpliedTransfers;
+import com.hedera.services.ledger.BalanceChange;
+import com.hedera.services.ledger.MerkleAccountScopedCheck;
+import com.hedera.services.ledger.TransactionalLedger;
+import com.hedera.services.ledger.properties.AccountProperty;
+import com.hedera.services.ledger.properties.NftProperty;
+import com.hedera.services.ledger.properties.TokenProperty;
+import com.hedera.services.ledger.properties.TokenRelProperty;
+import com.hedera.services.state.merkle.MerkleAccount;
+import com.hedera.services.state.merkle.MerkleToken;
+import com.hedera.services.state.merkle.MerkleTokenRelStatus;
+import com.hedera.services.state.merkle.MerkleUniqueToken;
+import com.hedera.services.state.submerkle.FcTokenAssociation;
+import com.hedera.services.store.models.NftId;
+import com.hedera.services.store.tokens.views.UniqTokenViewsManager;
+import com.hedera.services.txns.validation.OptionValidator;
+import com.hedera.services.utils.TxnAccessor;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.hederahashgraph.api.proto.java.TokenID;
+import com.hederahashgraph.api.proto.java.TransferList;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.hedera.services.exceptions.ValidationUtils.validateTrue;
+import static com.hedera.services.ledger.properties.AccountProperty.BALANCE;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.OK;
+
+public class TransferLogic {
+    private final MerkleAccountScopedCheck scopedCheck;
+    private final TransferList.Builder netTransfers = TransferList.newBuilder();
+    private final GlobalDynamicProperties dynamicProperties;
+    private final OptionValidator validator;
+    private UniqTokenViewsManager tokenViewsManager = null;
```

Comment:
This should be outside of this `TransferLogic` class. The transition logic should update the tokens view manager based on the side effects tracker